cmake-edit-core/src/main/java/ru/urururu/cmakeedit/core/checker/Checker.java

```diff
@@ -89,20 +88,82 @@ protected void init(Map<String, CommandSimulator> simulators) {
                             public List<SimulationState> getLoopStates() {
                                 return null;
                             }
-                        }, new SimulationState(functionBody, 0));
+                        }, new SimulationState(functionBody, 0, suspiciousPoints));
 
                         return functionSimulator.simulate(ctx, state, command);
                     });
+
+                    simulators.put("set", new CommandSimulator() {
+                        @Override
+                        public SimulationState simulate(CheckContext ctx, SimulationState state, CommandInvocationNode command) throws LogicalException {
+                            List<Node> arguments = command.getArguments();
+
+                            Iterator<Node> iterator = arguments.iterator();
+
+                            ArgumentNode variable;
+                            if (iterator.hasNext()) {
+                                variable = (ArgumentNode) iterator.next();
+                            } else {
+                                throw new LogicalException("wrong number of arguments", command, command);
+                            }
+
+                            String lastValue = null;
+                            while (iterator.hasNext()) {
+                                lastValue = state.getValue(iterator.next());
+                                state.processUsage(lastValue);
+                            }
+
+                            boolean parentScope = "PARENT_SCOPE".equals(lastValue);
+
+                            state.putValue(variable, command, parentScope);
+
+                            state.setPosition(state.getPosition() + 1);
+                            return state;
+                        }
+                    });
+
+                    simulators.put("unset", simulators.get("set")); // semantics are similar
+
+                    simulators.put("list", new CommandSimulator() {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Reduce this anonymous class number of lines from 25 to at most 20, or make it a named class. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1188)